cli/src/main/java/com/devonfw/tools/ide/tool/ToolCommandlet.java

```diff
@@ -177,47 +124,92 @@ protected final static String getToolWithEdition(String tool, String edition) {
+  }
+
+  /**
+   * Ensures the tool is installed and then runs this tool with the given arguments.
+   *
+   * @param processMode the {@link ProcessMode}. Should typically be {@link ProcessMode#DEFAULT} or {@link ProcessMode#BACKGROUND}.
+   * @param toolVersion the explicit {@link GenericVersionRange version} to run. Typically {@code null} to run the
+   *     {@link #getConfiguredVersion() configured version}. Otherwise, the specified version will be used (from the software repository, if not compatible).
+   * @param args the command-line arguments to run the tool.
+   */
+  public final void runTool(ProcessMode processMode, GenericVersionRange toolVersion, String... args) {
+
+    runTool(processMode, toolVersion, ProcessErrorHandling.THROW_CLI, args);
   }
 
   /**
-   * Method to be called for {@link #install(boolean)} from dependent {@link com.devonfw.tools.ide.commandlet.Commandlet}s. Additionally, contains the
-   * environmentContext of the tool
+   * Ensures the tool is installed and then runs this tool with the given arguments.
    *
-   * @param environmentContext - the environment context that can be used for the dependencies
-   * @return {@code true} if the tool was newly installed, {@code false} if the tool was already installed before and nothing has changed.
+   * @param processMode the {@link ProcessMode}. Should typically be {@link ProcessMode#DEFAULT} or {@link ProcessMode#BACKGROUND}.
+   * @param toolVersion the explicit {@link GenericVersionRange version} to run. Typically {@code null} to run the
+   *     {@link #getConfiguredVersion() configured version}. Otherwise, the specified version will be used (from the software repository, if not compatible).
+   * @param args the command-line arguments to run the tool.
+   */
+  public ProcessResult runTool(ProcessMode processMode, GenericVersionRange toolVersion, ProcessErrorHandling errorHandling, String... args) {
+
+    ProcessContext pc = this.context.newProcess().errorHandling(errorHandling);
+    install(true, pc);
+    configureToolBinary(pc, processMode, errorHandling);
+    configureToolArgs(pc, processMode, errorHandling, args);
+    return pc.run(processMode);
+  }
+
+  /**
+   * @param pc the {@link ProcessContext}.
+   * @param processMode the {@link ProcessMode}.
+   * @param errorHandling the {@link ProcessErrorHandling}.
+   */
+  protected void configureToolBinary(ProcessContext pc, ProcessMode processMode, ProcessErrorHandling errorHandling) {
+
+    pc.executable(Path.of(getBinaryName()));
+  }
+
+  /**
+   * @param pc the {@link ProcessContext}.
+   * @param processMode the {@link ProcessMode}.
+   * @param errorHandling the {@link ProcessErrorHandling}.
+   * @param args the command-line arguments to {@link ProcessContext#addArgs(Object...) add}.
    */
-  public boolean install(EnvironmentContext environmentContext) {
+  protected void configureToolArgs(ProcessContext pc, ProcessMode processMode, ProcessErrorHandling errorHandling, String... args) {
```

Comment:
Because this method is overridden by other commandlets where it is used and the information is required to make the proper decision. See e.g. Eclipse commandlet.